databuilder/transformer/elasticsearch_document_transformer.py

```diff
@@ -38,44 +39,9 @@ def transform(self, record):
 
-        if self.elasticsearch_resource_type.lower() not in \
-                ElasticsearchDocumentTransformer.RESOURCE_TYPE_MAPPING:
-            raise Exception('resource type needs to define in RESOURCE_TYPE_MAPPING')
-
-        if self.elasticsearch_resource_type == 'table':
-            elasticsearch_obj = TableESDocument(elasticsearch_index=self.elasticsearch_index,
-                                                elasticsearch_type=self.elasticsearch_type,
-                                                database=record.database,
-                                                cluster=record.cluster,
-                                                schema_name=record.schema_name,
-                                                table_name=record.table_name,
-                                                table_key=record.table_key,
-                                                table_description=record.table_description,
-                                                table_last_updated_epoch=record.table_last_updated_epoch,
-                                                column_names=record.column_names,
-                                                column_descriptions=record.column_descriptions,
-                                                total_usage=record.total_usage,
-                                                unique_usage=record.unique_usage,
-                                                tag_names=record.tag_names)
-        elif self.elasticsearch_resource_type == 'user':
-            elasticsearch_obj = UserESDocument(elasticsearch_index=self.elasticsearch_index,
-                                               elasticsearch_type=self.elasticsearch_type,
-                                               email=record.email,
-                                               first_name=record.first_name,
-                                               last_name=record.last_name,
-                                               name=record.name,
-                                               github_username=record.github_username,
-                                               team_name=record.team_name,
-                                               employee_type=record.employee_type,
-                                               manager_email=record.manager_email,
-                                               slack_id=record.slack_id,
-                                               is_active=record.is_active,
-                                               total_read=record.total_read,
-                                               total_own=record.total_own,
-                                               total_follow=record.total_follow,
-                                               )
-        else:
-            raise NotImplementedError()
+        elasticsearch_obj = self.model_class(elasticsearch_index=self.elasticsearch_index,
```

Comment:
what is the difference between `**vars(record)` and `**(record)`? And I think we should raise the exception in case the model_class can't  be instantiated correctly?

Reply:
If record is a dict object, then we can use `**(record)`. However `record` is a `[Neo4jDataResult`](https://github.com/lyft/amundsendatabuilder/blob/9a2616958c569a5feb611e30d863357c23124157/databuilder/transformer/elasticsearch_document_transformer.py#L38) object, so we have to use `**vars(record)`. See https://docs.python.org/3/library/functions.html#vars 

BTW, this is how I found `Neo4jDataResult` model has a mapping relationship with `TableESDocument` model

Reply:
Exception will be raised if there is anything wrong with model_class.

See added tests:
https://github.com/lyft/amundsendatabuilder/pull/79/files#diff-9bacf0d2063d7f5db9019e8c71c2a302R96
 and 
https://github.com/lyft/amundsendatabuilder/pull/79/files#diff-9bacf0d2063d7f5db9019e8c71c2a302R110

Reply:
Any more concern?

Reply:
lgtm